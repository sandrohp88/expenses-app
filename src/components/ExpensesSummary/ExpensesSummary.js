import React from 'react'
import getVisibleExpenses from '../VisibleExpenses/VisibleExpenses'
import { connect } from 'react-redux'
import { getTotalExpenses } from '../../selectors/expensesTotal'
import numeral from 'numeral'
export const ExpensesSummary = ({ expenses }) => (
  <div>
    <h2>
      Viewing {expenses.length} {expenses.length > 1 ? 'expenses' : 'expense'}{' '}
      totaling {numeral(getTotalExpenses(expenses) / 100).format('$0,000.00')}
    </h2>
  </div>
)

const mapStateToProps = state => ({
  expenses: getVisibleExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpensesSummary)
