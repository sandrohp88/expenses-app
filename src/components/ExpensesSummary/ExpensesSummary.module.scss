@import '../../goblalStyles/settings';
@import '../../goblalStyles/content-container';
@import '../../goblalStyles/button';

.pageHeader {
  background: #f7f7f7;
  padding: $l-size 0;
  margin-bottom: $l-size;

}
.pageHeader__actions {
  margin-top: $m-size;
}
.pageHeader__title {
  font-weight: 300;
  margin: 0;
  span {
    font-weight: 700;
  }
}
