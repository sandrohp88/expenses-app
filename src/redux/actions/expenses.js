import expensesRef from '../../firebase/firebase'
// ADD_EXPENSE
const addExpense = expense => ({
  type: 'ADD_EXPENSE',
  expense
})

// ADD_EXPENSE_ASYNC
const addExpenseAsync = (expenseData = {}) => {
  return async dispatch => {
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData
    const expense = { description, note, amount, createdAt }
    const response = await expensesRef.add(expense)
    return dispatch(addExpense({ id: response.id, ...expense }))
  }
}

// REMOVE_EXPENSE
const removeExpense = (id = '') => ({ type: 'REMOVE_EXPENSE', id })

// EDIT_EXPENSE
const editExpense = (id, updates) => ({ type: 'EDIT_EXPENSE', id, updates })

// SET_EXPENSES
const setExpenses = expenses => ({ type: 'SET_EXPENSES', expenses })

// GET_EXPENSES_ASYNC
const getExpensesAsync = () => {
  return async dispatch => {
    const response = await expensesRef.get()
    const expenses = []
    response.forEach(expense => {
      const {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
      } = expense.data()
      expenses.push({id:expense.id, description, note, amount, createdAt })
    })
    return dispatch(setExpenses(expenses))
  }
}

export {
  getExpensesAsync,
  addExpense,
  addExpenseAsync,
  removeExpense,
  editExpense,
  setExpenses
}
