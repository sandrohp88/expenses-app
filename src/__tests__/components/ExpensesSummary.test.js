import { ExpensesSummary } from '../../components/ExpensesSummary/ExpensesSummary'
import { shallow } from 'enzyme'
import expenses from '../../sampleData/expenses'
import React from 'react'
test('render without crash', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />)
  expect(wrapper).toMatchSnapshot()
})

test('render singular expense', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />)
  const singular = wrapper
    .find('h2')
    .text()
    .includes(' expense ')
  expect(singular).toBe(true)
})

test('render plural expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenses={[expenses[0], expenses[2]]} />
  )
  const plural = wrapper
    .find('h2')
    .text()
    .includes(' expenses ')
  expect(plural).toBe(true)
})
