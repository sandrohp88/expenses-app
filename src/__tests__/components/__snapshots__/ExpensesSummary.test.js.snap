// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crash 1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expenses
     
    totaling 
    $1,141.95
  </h2>
</div>
`;
